### PR TITLE
[SpellCheck] Refine spell check ci

### PR DESCRIPTION
**Pull request by 0mza987**
# Description

Summary the detected errors in the CI
image

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by wangchao1230 — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

File: .github/workflows/spell_check.yml [hunk 1]
```diff
@@ -26,5 +26,11 @@ jobs:
 
       - name: Run cspell
         run: |-
-          cspell lint "**"
-          echo -e "\e[91mNote: If the detected words are expected, add them into ".cspell.json" which is at the repo root folder.\e[0m" 
+          cspell lint "**" > cspell_check.log || true
```
Comment (D-W-) [inline .github/workflows/spell_check.yml:29]: Do we need to also redirect stderr to the log?

Comment (elliotzh) [inline .github/workflows/spell_check.yml:29]: will it always create a new file? if so, have we added the log file to gitignore?

**Review by elliotzh — APPROVED** [review]

File: .github/workflows/spell_check.yml (hunk 1 above)
Comment (0mza987) [reply .github/workflows/spell_check.yml:29]: Yes it always create this file and yes it's already in `.gitignore`

Comment (0mza987) [reply .github/workflows/spell_check.yml:29]: After testing, this command somehow outputs detected mis-spell words info into `stdout` and outputs the normal scan info into `stderr`, and we want to print the former info to summary.

**Review by brynn-code — APPROVED** [review]

**Pull request merged**